lib/chef/knife/bootstrap.rb

```diff
@@ -549,7 +549,28 @@ def run
 
         $stdout.sync = true
         register_client
-        connect!
+        begin
+          connect!
```

Comment:
Can we rescue this rescue block of  `connect!` itself?  Having the connection error handling in one place seems more manageable than splitting it across two.

Reply:
done